### PR TITLE
chore(http): improve security hooks

### PR DESCRIPTION
**Pull request by puzpuzpuz**
* Changes auth hooks for HTTP to make them suitable for HTTP basic auth
* Adds some tests with test-only authenticator implementations

Static content, metrics, and health check endpoints require authentication by default, but this may be disabled with the following config:
```ini
http.static.authentication.required=false
http.health.check.authentication.required=false
```

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 59 / 59 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/auth/EllipticCurveLineAuthenticatorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGWireServer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/DefaultLineAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/StaticPgWireAuthenticatorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/DefaultPgWireAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpAuthenticator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpAuthenticatorFactory.java|3|3|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 85 / 85 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/EllipticCurveLineAuthenticatorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGWireServer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/DefaultLineAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/StaticPgWireAuthenticatorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/DefaultPgWireAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpAuthenticator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/Services.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpAuthenticatorFactory.java|3|3|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 85 / 85 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/EllipticCurveLineAuthenticatorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGWireServer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/DefaultLineAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/StaticPgWireAuthenticatorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/DefaultPgWireAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpAuthenticator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/Services.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpAuthenticatorFactory.java|3|3|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 164 / 164 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/EllipticCurveLineAuthenticatorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGWireServer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/DefaultLineAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/std/Chars.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/std/Encoding.java|64|64|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/DefaultPgWireAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/UserDatabasePgWireAuthenticatorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpAuthenticator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/AuthUtils.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/Services.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpAuthenticatorFactory.java|3|3|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 214 / 223 (95.96%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/DirectByteCharSink.java|29|37|78.38%|
|:large_blue_circle:|io/questdb/std/Encoding.java|84|85|98.82%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/EllipticCurveLineAuthenticatorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGWireServer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/DefaultLineAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/std/Chars.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/CleartextPasswordPgWireAuthenticator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/DefaultPgWireAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/UserDatabasePgWireAuthenticatorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpAuthenticator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/AuthUtils.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/Services.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpAuthenticatorFactory.java|3|3|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 214 / 223 (95.96%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/DirectByteCharSink.java|29|37|78.38%|
|:large_blue_circle:|io/questdb/std/Encoding.java|84|85|98.82%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/EllipticCurveLineAuthenticatorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGWireServer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/DefaultLineAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/std/Chars.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/CleartextPasswordPgWireAuthenticator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/DefaultPgWireAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/UserDatabasePgWireAuthenticatorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpAuthenticator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/AuthUtils.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/Services.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpAuthenticatorFactory.java|3|3|100.00%|

**Comment by puzpuzpuz** [comment]
cc @mariusgheorghies who might be interested in the default health check protection and the new `http.health.check.authentication.required` prop.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 138 / 147 (93.88%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/DirectByteCharSink.java|29|37|78.38%|
|:large_blue_circle:|io/questdb/std/Chars.java|20|21|95.24%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/EllipticCurveLineAuthenticatorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGWireServer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/DefaultLineAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/CleartextPasswordPgWireAuthenticator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/DefaultPgWireAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/UserDatabasePgWireAuthenticatorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpAuthenticator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/Services.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpAuthenticatorFactory.java|3|3|100.00%|

File: core/src/main/java/io/questdb/std/Chars.java [hunk 1]
```diff
@@ -1054,6 +1058,36 @@ private static int[] base64CreateInvertedAlphabet(char[] alphabet) {
         return inverted;
     }
 
+    private static void base64Decode(CharSequence encoded, DirectByteCharSink target, int[] invertedAlphabet) {
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/std/Chars.java:1061]: Good point. Updated with the same algo in dee5d58668a9b39ab2c38b91832166b0fb44d0c2

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 155 / 164 (94.51%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/DirectByteCharSink.java|29|37|78.38%|
|:large_blue_circle:|io/questdb/std/Chars.java|33|34|97.06%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/StaticUsernamePasswordMatcher.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/EllipticCurveLineAuthenticatorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGWireServer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/auth/DefaultLineAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/UsernamePasswordPgWireAuthenticatorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpResponseSink.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/CleartextPasswordPgWireAuthenticator.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/DefaultPgWireAuthenticatorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpAuthenticator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/Services.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpAuthenticatorFactory.java|3|3|100.00%|

**Review by jerrinot — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
@jerrinot thanks for the review!

**Pull request merged**